simulariumio/filters/add_agents_filter.py

```diff
@@ -33,5 +33,5 @@ def apply(self, data: TrajectoryData) -> TrajectoryData:
         Add the given agents to each frame of the simularium data
         """
         print("Filtering: add agents -------------")
-        data.agent_data.append_agents(self.new_agent_data)
+        data.agent_data = data.agent_data.append_agents(self.new_agent_data)
```

Comment:
ok I fixed this, with the `append_agents()` function on `AgentData` I couldn't set the object itself to the new value, requiring it to be awkward and non-intuitive like this (or the tests fail). So I moved `append_agents()` to `TrajectoryData` because it can set it's `agent_data`, and also made the function names and comments clear about when an `AgentData` is getting copied. It doesn't need to be copied here because it should already be copied by `TrajectoryConverter` before it called `apply()` on this filter.